core/tests/protractor_desktop/topicAndStoryEditor.js

```diff
@@ -197,3 +197,160 @@ describe('Topic editor functionality', function() {
+    storyEditorPage.get(storyId);
+  });
+
+  it('should create a basic chapter.', function() {
+    storyEditorPage.createInitialChapter('Chapter 1');
+    storyEditorPage.setChapterExplorationId(dummyExplorationIds[0]);
+    storyEditorPage.changeNodeOutline(forms.toRichText('First outline'));
+    storyEditorPage.saveStory('First save');
+  });
+
+  it('should add one more chapter to the story', function() {
+    storyEditorPage.createNewDestinationChapter('Chapter 2');
+    storyEditorPage.navigateToChapterByIndex(1);
+    storyEditorPage.changeNodeOutline(forms.toRichText('Second outline'));
+    storyEditorPage.setChapterExplorationId(dummyExplorationIds[1]);
+    storyEditorPage.saveStory('First save');
+  });
+
+  it('should try to add one more chapter with existing exploration',
+    function() {
+      storyEditorPage.navigateToChapterByIndex(1);
+      storyEditorPage.createNewDestinationChapter('Chapter 3');
+      storyEditorPage.navigateToChapterByIndex(2);
+      storyEditorPage.setChapterExplorationId(dummyExplorationIds[1]);
+      storyEditorPage.expectExplorationIdAlreadyExistWarning();
+      allowedError = ['The given exploration already exists in the story.'];
+    }
+  );
+
+  it('should add one more chapter and change the chapters sequences',
+    function() {
+      storyEditorPage.navigateToChapterByIndex(1);
+      storyEditorPage.createNewDestinationChapter('Chapter 3');
+      storyEditorPage.navigateToChapterByIndex(2);
+      storyEditorPage.setChapterExplorationId(dummyExplorationIds[2]);
+      storyEditorPage.selectInitialChapterByName('Chapter 2');
+
+      // Now Chapter 2 is the initial chapter and its destination is
+      // Chapter 3. Make Chapter 2's destination to be Chapter 1
+      storyEditorPage.navigateToChapterByIndex(0);
+      storyEditorPage.removeDestination();
+      storyEditorPage.selectDestinationChapterByName('Chapter 1');
+      storyEditorPage.expectDestinationToBe('Chapter 1');
+
+      // Make chapter 1's destination to be Chapter 3
+      storyEditorPage.navigateToChapterByIndex(1);
+      storyEditorPage.selectDestinationChapterByName('Chapter 3');
+      storyEditorPage.expectDestinationToBe('Chapter 3');
+    }
+  );
+
+  it('should add one prerequisite and acquired skill to chapter 1', function() {
+    storyEditorPage.expectAcquiredSkillDescriptionCardNumber(0);
+    storyEditorPage.expectPrerequisiteSkillDescriptionCardNumber(0);
+    storyEditorPage.addAcquiredSkill(dummySkills[0]);
+    storyEditorPage.expectAcquiredSkillDescriptionCardNumber(1);
+    storyEditorPage.addPrerequisiteSkill(dummySkills[1]);
+    storyEditorPage.expectPrerequisiteSkillDescriptionCardNumber(1);
+    storyEditorPage.saveStory('Save');
+  });
+
+  it('should fail to add one prerequisite skill which is already added as' +
+    ' acquired skill', function() {
+    storyEditorPage.addAcquiredSkill(dummySkills[1]);
+    storyEditorPage.expectSaveStoryDisabled();
```

Comment:
Can we check that the user is told what's wrong? (It'd be super confusing for the save button to be disabled with no explanation as to why)

Reply:
There is only one warning item in the warning list, not sure how to check that though.

Reply:
@cuichenli There is an HTML element that displays the warning to the user, right? Can we just check whether that element is visible and has the appropriate text?

Reply:
Looks great, thanks!